fpdf/fpdf.py

```diff
@@ -862,6 +864,52 @@ def set_line_width(self, width):
         if self.page > 0:
             self._out(f"{width * self.k:.2f} w")
 
+    @contextmanager
+    @check_page
+    def drawing_context(self, dump_render_tree=None):
+        """
+        Create a context for drawing paths on the current page.
+
+        If this context manager is called again inside of an active context, it will
+        simply return the existing drawing context, as base drawing contexts cannot be
+        nested.
+
+        Args:
+            dump_render_tree (TextIO): print a pretty tree of all items to be rendered
+                to the provided stream. To store the output in a string, use io.StringIO.
+        """
+
+        if self._current_draw_context is not None:
+            yield self._current_draw_context
+        else:
+            context = drawing.DrawingContext()
+            self._current_draw_context = context
+            try:
+                yield context
+            finally:
+                self._current_draw_context = None
+
+            rendered = context.render(drawing.Point(0, 0), self.k)
+            if dump_render_tree is not None:
+                context.dump_render_tree(dump_render_tree)
+            self._out(rendered)
+
+    @contextmanager
+    def path(self, paint_style, x, y):
```

Comment:
Nice method: simple, clear, efficient.

What do you think about having this `paint_style` argument being the 3rd one, with a default value of `PaintStyle.stroke`. I'd like users to be able to easily experiment with this without having to look at the `drawing` module internals.

Reply:
I have addressed this by introducing a default of `PaintStyle.AUTO` which is resolved to a concrete paint style upon rendering by inspecting the existence of other style attributes, such as the stroke width/color and fill color of the path. I still need to make sure this produces reasonable results when no style attributes are set, but this is mostly solved.